pkg/xds/envoy/listeners/v2/inbound_listener_configurer.go

```diff
@@ -3,21 +3,32 @@ package v2
 import (
 	v2 "github.com/envoyproxy/go-control-plane/envoy/api/v2"
 	envoy_core "github.com/envoyproxy/go-control-plane/envoy/api/v2/core"
+
+	mesh_core "github.com/kumahq/kuma/pkg/core/resources/apis/mesh"
 )
 
 type InboundListenerConfigurer struct {
 	ListenerName string
 	Address      string
 	Port         uint32
+	Protocol     mesh_core.Protocol
 }
 
 func (c *InboundListenerConfigurer) Configure(l *v2.Listener) error {
 	l.Name = c.ListenerName
 	l.TrafficDirection = envoy_core.TrafficDirection_INBOUND
+	var listenerProtocol envoy_core.SocketAddress_Protocol
+	switch c.Protocol {
+	case mesh_core.ProtocolUDP:
+		listenerProtocol = envoy_core.SocketAddress_UDP
+		l.ReusePort = true
+	case mesh_core.ProtocolTCP:
+		listenerProtocol = envoy_core.SocketAddress_TCP
```

Comment:
Nit: Maybe return an error if it's neither TCP nor UDP to make configuration/dev errors more obvious to users (atm this falls back to TCP which is ok 99% but can lead to hard to trackdown bugs).